File: Lib/Telerik/Examples/QuickStart/PropertyGrid/FirstLook/RadElementWrapper.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.ComponentModel;
using System.Windows.Forms;
using System.Drawing;
using Telerik.WinControls.Primitives;
using Telerik.WinControls.UI;
using Telerik.WinControls;

namespace Telerik.Examples.WinControls.PropertyGrid.FirstLook
{
    public class RadElementWrapper
    {
        RadItem element;
        SurfaceObject owner;

        public RadElementWrapper(SurfaceObject owner, RadItem element)
        {
            this.element = element;
            this.owner = owner;
        }

        #region Properties

        [Description("Gets or sets the text displayed in the control.")]
        [Category("Text")]
        [DefaultValue("RadPropertyGrid")]
        [Browsable(true)]
        public string Text
        {
            get
            {
                return this.element.Text;
            }
            set
            {
                this.element.Text = value;
            }
        }

        [Description("Gets or sets the size of the text displayed in the control.")]
        [Category("Text")]
        [DefaultValue(12)]
        [Browsable(true)]
        public float FontSize
        {
            get
            {
                return this.element.Font.Size;
            }
            set
            {
                this.element.Font = new Font(this.element.Font.FontFamily, value);
            }
        }

        [Description("Gets or sets the width of the control border.")]
        [Category("Border")]
        [DefaultValue(2.0f)]
        [Browsable(true)]
        public float BorderWidth
        {
            get
            {
                LightVisualElement lve = element as LightVisualElement;
                if (lve != null)
                {
                    return lve.BorderWidth;
                }
                BorderPrimitive border = element.FindDescendant<BorderPrimitive>();
                if (border != null)
                {
                    return border.Width;
                }
                return 1;
            }
            set
            {
                LightVisualElement lve = element as LightVisualElement;
                if (lve != null)
                {
                    lve.BorderWidth = value;
                }
                BorderPrimitive border = element.FindDescendant<BorderPrimitive>();
                if (border != null)
                {
                    border.Width = value;
                }
            }
        }

        [Description("Gets or sets the color of the control border.")]
        [Category("Border")]
        [DefaultValue(null)]
        [Browsable(true)]
        public Color BorderColor
        {
            get
            {
                LightVisualElement lve = element as LightVisualElement;
                if (lve != null)
                {
                    return lve.BorderColor;
                }
                BorderPrimitive border = element.FindDescendant<BorderPrimitive>();
                if (border != null)
                {
                    return border.ForeColor;
                }
                return Color.Black;
            }
            set
            {
                LightVisualElement lve = element as LightVisualElement;
                if (lve != null)
                {
                    lve.BorderColor = value;
                }
                BorderPrimitive border = element.FindDescendant<BorderPrimitive>();
                if (border != null)
                {
                    border.Width = Math.Max(1, border.Width);
                    border.BoxStyle = BorderBoxStyle.SingleBorder;
                    border.GradientStyle = GradientStyles.Solid;
                    border.Visibility = ElementVisibility.Visible;
                    border.ForeColor = value;
                    border.ShouldPaint = true;
                }
            }
        }

        [Description("Gets or sets the color of the fill of the control border.")]
        [Category("Background")]
        [DefaultValue(null)]
        [Browsable(true)]
        public Color BackColor
        {
            get
            {
                LightVisualElement lve = element as LightVisualElement;
                if (lve != null)
                {
                    return lve.BackColor;
                }
                FillPrimitive fill = element.FindDescendant<FillPrimitive>();
                if (fill != null)
                {
                    return fill.BackColor;
                }
                return Color.Black;
            }
            set
            {
                LightVisualElement lve = element as LightVisualElement;
                if (lve != null)
                {
                    lve.BackColor = value;
                }
                FillPrimitive fill = element.FindDescendant<FillPrimitive>();
                if (fill != null)
                {
                    fill.Visibility = ElementVisibility.Visible;
                    fill.ShouldPaint = true;
                    fill.GradientStyle = GradientStyles.Solid;
                    fill.BackColor = value;
                }
            }
        }

        [Description("Gets or sets the color of the text of the control border.")]
        [Category("Text")]
        [DefaultValue(null)]
        [Browsable(true)]
        public Color ForeColor
        {
            get
            {
                return this.element.ForeColor;
            }
            set
            {
                this.element.ForeColor = value;
            }
        }

        [Description("Gets or sets location of the control.")]
        [Category("Layout")]
        [DefaultValue(null)]
        [Browsable(true)]
        public Point Location
        {
            get
            {
                return this.owner.Offset;
            }
            set
            {
                this.owner.Offset = value;
            }
        }

        [Description("Gets or sets padding of the control.")]
        [Category("Layout")]
        [DefaultValue(null)]
        [Browsable(true)]
        public Padding Padding
        {
            get
            {
                return this.element.Padding;
            }
            set
            {
                this.element.Padding = value;
                this.owner.InvalidateMeasure(true);
            }
        }

        #endregion
    }
}

File: Lib/Telerik/Examples/QuickStart/PropertyGrid/FirstLook/RadElementWrapper.vb
Imports System.Text
Imports System.ComponentModel
Imports Telerik.WinControls.Primitives
Imports Telerik.WinControls.UI
Imports Telerik.WinControls

Namespace Telerik.Examples.WinControls.PropertyGrid.FirstLook
    Public Class RadElementWrapper
        Private element As RadItem
        Private owner As SurfaceObject

        Public Sub New(ByVal owner As SurfaceObject, ByVal element As RadItem)
            Me.element = element
            Me.owner = owner
        End Sub

#Region "Properties"

        <Description("Gets or sets the text displayed in the control."), Category("Text"), DefaultValue("RadPropertyGrid"), Browsable(True)> _
        Public Property Text() As String
            Get
                Return Me.element.Text
            End Get
            Set(ByVal value As String)
                Me.element.Text = value
            End Set
        End Property

        <Description("Gets or sets the size of the text displayed in the control."), Category("Text"), DefaultValue(12), Browsable(True)> _
        Public Property FontSize() As Single
            Get
                Return Me.element.Font.Size
            End Get
            Set(ByVal value As Single)
                Me.element.Font = New Font(Me.element.Font.FontFamily, value)
            End Set
        End Property

        <Description("Gets or sets the width of the control border."), Category("Border"), DefaultValue(2.0F), Browsable(True)> _
        Public Property BorderWidth() As Single
            Get
                Dim lve As LightVisualElement = TryCast(element, LightVisualElement)
                If lve IsNot Nothing Then
                    Return lve.BorderWidth
                End If
                Dim border As BorderPrimitive = element.FindDescendant(Of BorderPrimitive)()
                If border IsNot Nothing Then
                    Return border.Width
                End If
                Return 1
            End Get
            Set(ByVal value As Single)
                Dim lve As LightVisualElement = TryCast(element, LightVisualElement)
                If lve IsNot Nothing Then
                    lve.BorderWidth = value
                End If
                Dim border As BorderPrimitive = element.FindDescendant(Of BorderPrimitive)()
                If border IsNot Nothing Then
                    border.Width = value
                End If
            End Set
        End Property

        <Description("Gets or sets the color of the control border.")> _
        <Category("Border")> _
        <Browsable(True)> _
        Public Property BorderColor() As Color
            Get
                Dim lve As LightVisualElement = TryCast(element, LightVisualElement)
                If lve IsNot Nothing Then
                    Return lve.BorderColor
                End If
                Dim border As BorderPrimitive = element.FindDescendant(Of BorderPrimitive)()
                If border IsNot Nothing Then
                    Return border.ForeColor
                End If
                Return Color.Black
            End Get
            Set(ByVal value As Color)
                Dim lve As LightVisualElement = TryCast(element, LightVisualElement)
                If lve IsNot Nothing Then
                    lve.BorderColor = value
                End If
                Dim border As BorderPrimitive = element.FindDescendant(Of BorderPrimitive)()
                If border IsNot Nothing Then
                    border.Width = Math.Max(1, border.Width)
                    border.BoxStyle = BorderBoxStyle.SingleBorder
                    border.GradientStyle = GradientStyles.Solid
                    border.Visibility = ElementVisibility.Visible
                    border.ForeColor = value
                    border.ShouldPaint = True
                End If
            End Set
        End Property

        <Description("Gets or sets the color of the fill of the control border.")> _
              <Category("Background")> _
              <Browsable(True)> _
        Public Property BackColor() As Color
            Get
                Dim lve As LightVisualElement = TryCast(element, LightVisualElement)
                If lve IsNot Nothing Then
                    Return lve.BackColor
                End If
                Dim fill As FillPrimitive = element.FindDescendant(Of FillPrimitive)()
                If fill IsNot Nothing Then
                    Return fill.BackColor
                End If
                Return Color.Black
            End Get
            Set(ByVal value As Color)
                Dim lve As LightVisualElement = TryCast(element, LightVisualElement)
                If lve IsNot Nothing Then
                    lve.BackColor = value
                End If
                Dim fill As FillPrimitive = element.FindDescendant(Of FillPrimitive)()
                If fill IsNot Nothing Then
                    fill.Visibility = ElementVisibility.Visible
                    fill.ShouldPaint = True
                    fill.GradientStyle = GradientStyles.Solid
                    fill.BackColor = value
                End If
            End Set
        End Property

        <Description("Gets or sets the color of the text of the control border.")> _
                <Category("Text")> _
                <Browsable(True)> _
        Public Property ForeColor() As Color
            Get
                Return Me.element.ForeColor
            End Get
            Set(ByVal value As Color)
                Me.element.ForeColor = value
            End Set
        End Property

        <Description("Gets or sets location of the control.")> _
      <Category("Layout")> _
      <Browsable(True)> _
        Public Property Location() As Point
            Get
                Return Me.owner.Offset
            End Get
            Set(ByVal value As Point)
                Me.owner.Offset = value
            End Set
        End Property

        <Description("Gets or sets padding of the control.")> _
              <Category("Layout")> _
              <Browsable(True)> _
        Public Property Padding() As Padding
            Get
                Return Me.element.Padding
            End Get
            Set(ByVal value As Padding)
                Me.element.Padding = value
                Me.owner.InvalidateMeasure(True)
            End Set
        End Property

#End Region
    End Class
End Namespace
